RFS/src/main/java/org/opensearch/migrations/bulkload/common/LuceneDocumentsReader.java

```diff
@@ -105,6 +109,28 @@ public Flux<RfsLuceneDocument> readDocuments(int startSegmentIndex, int startDoc
         });
     }
 
+    /**
+     * We need to ensure a stable ordering of segments so we can start reading from a specific segment and document id.
+     * To do this, we sort the segments by their name.
+     */
```

Comment:
This won't work, IIRC - Mikayla was doing something like this and not everything will be a SegmentReader.  She was originally using 'name', but it wasn't always present.  We need some other way to put these into a canonical order - anything that's stored in the metadata that's stable and unique would work.  Size wouldn't be unique, so that could be a problem...

Does the metadata not have these in a stable order that's pulled the same way every time?

Reply:
@AndreKurait and I found a simple solution here.